### PR TITLE
Library version

### PR DESCRIPTION
**Pull request by xuefei-wang**
I have updated the repo so that we can take numpy arrays and run a simple `predict` function, instead of expecting users to format the data into zarrs. This implementation addresses the request in [issue ticket #6](https://github.com/vanvalenlab/deepcell-types/issues/6). 

This is a very initial basic implementation and more refinements are definitely needed. I've outlined some TODOs below for future reference: 
- [ ] Add argparser to demo/inference.py 
- [ ] `orig_ct` in `patch_generator` is set to `Unknown` - we should remove this arg since we don't expect users to have it
- [ ] Clean up `deepcell_types/dct_kit` folder; merge `data/` and `deepcell_types/dct_kit/config/`
- [ ] Add dtype check of input data
- [ ] Update documentation and README accordingly
- [ ] Decide if we still want to add instructions about how to use the Docker version; if not, remove Dockerfile
- [ ] Do proper packaging by updating project.toml / checking python versions & package versions / ...
- [ ] Add instructions on how to obtain DeepSeek embeddings for cell types and markers
- [ ] Add more channel name mappings ( i.e. variations of a certain marker name -> the standard name )

### CODE REVIEW

**Comment by xuefei-wang** [comment]
Adding reviewers' requests here as well: 
- [ ] Add detailed instructions for setting up the environment
- [ ] Provide clear documentation of dataset access and step-by-step instructions on how to run the workflow
- [ ] Outline the limitations of the approach ( i.e. under which context it is most effective and when it might not be suitable )
- [ ] Add a demo ( jupyter notebooks or video tutorials ), including running segmentation, phenotyping and visualization

**Comment by rossbar** [comment]
Okay I've pushed up the changes I made to get things running on a data-local machine. 

The most important change was to force `num_workers=1` in the DataLoader - more workers is the main cause of memory blowup (to be addressed separately down the road - avoiding premature optimization for now).

File: deepcell_types/dct_kit/image_funcs.py [hunk 1]
```diff
@@ -130,25 +129,17 @@ def patch_generator(raw, mask, mpp, dct_config, final_q, cell_index=None, cell_t
         anti_aliasing=False,
     ).astype(np.int32)
 
-    raw = raw / final_q[None, None, :]
-    # if final_q has 0, then raw will have nan, replace nan with 0
-    raw = np.nan_to_num(raw, nan=0)
-
-    raw = np.clip(raw, 0, 5)
+    raw = percentile_threshold(raw, percentile=dct_config.PERCENTILE_THRESHOLD)
+    raw = np.clip(raw, 0, 5) # clip to remove the extreme values
```
Comment (rossbar) [inline deepcell_types/dct_kit/image_funcs.py:133]: This strikes me as problematic - AFAICT, the `raw` here undergoes no normalization at all before the clipping... this would mean that there is no attempt to standardize the numeric values of the pixels for the input datasets. Maybe there is an assumed normalization step prior to this (that either isn't taking place or is and I missed it!)

Comment (xuefei-wang) [reply deepcell_types/dct_kit/image_funcs.py:133]: Good catch! Thanks for noticing it. Yes, the previous version included a normalization step and clipping is legacy operation because the group-based statistics does not help normalize every FOV to be between 0 and 1. 

In this version, we should not be doing clipping; instead, we should do normalization directly. 
I just had a discussion with @raffeyC over slack about it.
Raffey, could you update both this repo and the [pipeline](https://github.com/xuefei-wang/deepcelltypes-kit/) of training data generation?

Comment (rossbar) [reply deepcell_types/dct_kit/image_funcs.py:133]: BTW I'd like to credit @raffeyC with catching this, so thanks for that!
